pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -725,7 +725,6 @@ func (r *ReconcilePerconaServerMongoDB) reconcileStatefulSet(arbiter bool, cr *a
 	if err != nil {
 		return nil, fmt.Errorf("create StatefulSet.Spec %s: %v", sfs.Name, err)
 	}
-	sfsSpec.Template.Annotations = sfs.Spec.Template.Annotations
```

Comment:
why this line is removed?

Reply:
because sfs.Spec.Template.Annotations is nill

Reply:
if we remove this line, users rotation can become broken https://github.com/percona/percona-server-mongodb-operator/commit/04e95295

Reply:
reworked